### PR TITLE
Fix BNN import

### PR DESCRIPTION
**Pull request by robwa**
tested with Terra BNN files

### CODE REVIEW

File: app/models/supplier.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ def update_articles_from_file(file, **opts)
         article.destroy && outlisted_counter += 1
 
       # remember special info for article; store data to allow article after its special
-      elsif status == :special
+      elsif status == :special and article
```
Comment (wvengen) [inline app/models/supplier.rb]: for the boolean operator, please use `&&`

**Comment by wvengen** [comment]
Were these two issues the reason for https://github.com/foodcoops/foodcoops.net/issues/4#issuecomment-348126088 not happening?

File: app/models/supplier.rb (hunk 1 above)
Comment (robwa) [reply app/models/supplier.rb]: I used the same operator as it was used in line [65](https://github.com/robwa/sharedlists/blob/7c35a7dd3f1b57aa87adb2a0f0b707e550993fed/app/models/supplier.rb#L65) and [70](https://github.com/robwa/sharedlists/blob/7c35a7dd3f1b57aa87adb2a0f0b707e550993fed/app/models/supplier.rb#L70) above. Change only this one?

**Comment by robwa** [comment]
> Were these two issues the reason for foodcoops/foodcoops.net#4 (comment) not happening?

Yes.

File: app/models/supplier.rb (hunk 1 above)
Comment (wvengen) [reply app/models/supplier.rb]: Wow, I missed that. Thanks for keeping to local style! :)
I'd say it's better to change all of them, but feel free to take your pick. Happy to merge when ready.

Comment (robwa) [reply app/models/supplier.rb]: I change the operators in a separate commit, to make clear that it's only refactoring.

**Pull request merged**